code/game/objects/items/devices/taperecorder.dm

```diff
@@ -345,6 +334,7 @@
 	var/unspooled = FALSE
 	var/list/icons_available = list()
 	var/radial_icon_file = 'icons/hud/radial_tape.dmi'
+	var/firstFlip = TRUE
```

Comment:
`first_flip`--snake_case only

Reply:
Totally up for learning a new way to do things, but I hope you don't mind my questions-- I'm actively trying to learn and get a deeper understanding of the processes involved. 

     The if changes reverted are for the purpose of making the function more easily understood as you noted, but doesn't that come at the cost of the codes efficiency? Or does nitpicking not do much to help Byond? I've heard it was a finicky engine.

     The decision of no more verbs, is this because verbs have a detrimental impact on performance? Or because of a more firm vision of unified design/intuition to lead the future development? On that note, regarding no verbs, are the altclick and other hotkeys acceptable still?

Thank you so much for taking your time to review this. I hope to hear back! Feel free to add me on Discord if you prefer NOXA#0082

Reply:
1. If it's less efficient, then it's a matter of microseconds--we optimize for readability outside of hot paths. 

Quoting the often-cut-off Donald Knuth:

> "Programmers waste enormous amounts of time thinking about, or worrying about, the speed of noncritical parts of their programs, and these attempts at efficiency actually have a strong negative impact when debugging and maintenance are considered. We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. Yet we should not pass up our opportunities in that critical 3%." 

That 3% is in places like atmos etc, but not something like tape recorders.

2. Verbs are clunky UX. Right-clicking tiles is often slow, you have to *know* the item has a verb, and players no longer expect it. Hotkeys are completely accepted. Contextual screentips exist so people can simply hover over the item and learn its hotkeys.

Reply:
> Quoting the often-cut-off Donald Knuth:

That was really good. Helped me see things in a new way and it has been kind of popping into my head every so often every time I try and learn and take on a new project — thanks for the wisdom, for real! 

Now, when you say contextual screen tips you're referring primarily to the item description and what the item says on the chatbox when a specific action is done, correct? So you're saying I should allude to features either directly or indirectly as to help give intuition to the player?